File: src/main/java/com/nickcoblentz/montoya/settings/LongExtensionSetting.java
package com.nickcoblentz.montoya.settings;

import burp.api.montoya.MontoyaApi;

public class LongExtensionSetting extends GenericExtensionSetting<Long> {

    public LongExtensionSetting(MontoyaApi api, String name, String storageKey, Long defaultValue, ExtensionSettingSaveLocation defaultSaveLocation) {
        super(api, name, storageKey, defaultValue, defaultSaveLocation);
    }

    @Override
    public void loadFromProject() {
        setCurrentValue(getApi().persistence().extensionData().getLong(getStorageKey()));
    }

    @Override
    public void loadFromPreferences() {
        setCurrentValue(getApi().persistence().preferences().getLong(getStorageKey()));
    }

    @Override
    public Long parseString(String value) {
        return Long.valueOf(value);
    }

    public void setCurrentValue(String value) {
        if(value!=null)
            setCurrentValue(parseString(value));
    }

    @Override
    public void saveToProject() {
        getApi().persistence().extensionData().setLong(getStorageKey(),getCurrentValue().longValue());
    }

    @Override
    public void saveToPreferences() {
        getApi().persistence().preferences().setLong(getStorageKey(),getCurrentValue().longValue());
    }
}

File: src/main/java/com/nickcoblentz/montoya/settings/LongExtensionSetting.kt
package com.nickcoblentz.montoya.settings

import burp.api.montoya.MontoyaApi

class LongExtensionSetting(
    api: MontoyaApi,
    name: String,
    storageKey: String,
    defaultValue: Long,
    defaultSaveLocation: ExtensionSettingSaveLocation
) :
    GenericExtensionSetting<Long>(api, name, storageKey, defaultValue, defaultSaveLocation) {
    override fun loadFromProject() {
        currentValue = api.persistence().extensionData().getLong(storageKey)
    }

    override fun loadFromPreferences() {
        currentValue = api.persistence().preferences().getLong(storageKey)
    }

    override fun parseString(value: String): Long {
        return value.toLong()
    }

    override fun setCurrentValue(value: String?) {
        if (value != null) currentValue = parseString(value)
    }

    override fun saveToProject() {
        api.persistence().extensionData().setLong(storageKey, currentValue)
    }

    override fun saveToPreferences() {
        api.persistence().preferences().setLong(storageKey, currentValue)
    }
}